src/sdk/helpers/requestsManager.ts

```diff
@@ -0,0 +1,187 @@
+                }
+            }
+        }
+    }
+
+    checkReferrerOrigin(): void {
+        const docReferrer = document.referrer;
+
+        if (!docReferrer) throwError('No referrer found', SdkErrors.ReferrerEmpty);
+
+        const referrerOrigin = new URL(docReferrer).origin;
+
+        const myRequest = this.getLoginRequestsOrThrow().find(
+            (request) => request.getPayload().origin === referrerOrigin
+        );
+
+        if (!myRequest) {
+            const msg =
+                `No origins from: ${this.getLoginRequestsOrThrow().map((r) => r.getPayload().origin)} ` +
+                `match referrer: ${referrerOrigin}`;
+
+            throwError(msg, SdkErrors.WrongOrigin);
+        }
+    }
+
+    getLoginRequests(): LoginRequest[] | undefined {
+        const response = this.requests.filter((request) => request instanceof LoginRequest);
+
+        return response.length > 0 ? response : undefined;
+    }
+
+    getLoginRequestsOrThrow(): LoginRequest[] {
+        const response = this.getLoginRequests();
+
+        if (!response) {
+            throwError('No LoginRequests found', SdkErrors.RequestsNotFound);
+        }
+
+        return response;
+    }
+
+    getDataSharingRequests(): DataSharingRequest[] | undefined {
+        const response = this.requests.filter((request) => request instanceof DataSharingRequest);
+
+        return response.length > 0 ? response : undefined;
+    }
+
+    getDataSharingRequestsOrThrow(): DataSharingRequest[] {
+        const response = this.getDataSharingRequests();
+
+        if (!response) {
+            throwError('No DataSharingRequests found', SdkErrors.RequestsNotFound);
+        }
+
+        return response;
+    }
+
+    getLoginRequestWithSameOriginOrThrow(): LoginRequest {
+        const myOrigin = window.location.origin;
+        const response = this.getLoginRequestsOrThrow().find((request) => request.getPayload().origin === myOrigin);
+
+        if (!response) {
+            throwError(`No origin from ${myOrigin} found`, SdkErrors.OriginNotFound);
+        }
+
+        return response;
+    }
+
+    getLoginRequestWithDifferentOriginOrThrow(): LoginRequest {
+        const myOrigin = window.location.origin;
+        const response = this.getLoginRequestsOrThrow().find((request) => request.getPayload().origin !== myOrigin);
+
+        if (!response) {
+            throwError(`No origin different from ${myOrigin} found`, SdkErrors.OriginNotFound);
+        }
+
+        return response;
+    }
+
+    getRequestsSameOriginOrThrow(): WalletRequest[] {
+        const loginRequest = this.getLoginRequestWithSameOriginOrThrow();
+        const issuer = loginRequest.getIssuer();
+
+        // TODO should maybe add origin to the DataRequest object instead of using the issuer?
+        const dataSharingRequest = this.requests.find(
+            (request) => request instanceof DataSharingRequest && request.getIssuer() === issuer
```

Comment:
and here we can get the origin from request?

Reply:
yes correct